add_target.lua

```diff
@@ -2,110 +2,107 @@ local S = minetest.get_translator("travelnet")
-       and not( travelnet.allow_attach( player_name, owner_name, network_name ))) then
-
-      travelnet.show_message( pos, player_name, S("Error"),
-	S("You do not have the travelnet_attach priv which is required to attach your box to "..
-	"the network of someone else. Aborting."));
-      return;
-   end
-
-   -- first one by this player?
-   if( not( travelnet.targets[ owner_name ] )) then
-      travelnet.targets[       owner_name ] = {};
-   end
-
-   -- first station on this network?
-   if( not( travelnet.targets[ owner_name ][ network_name ] )) then
-      travelnet.targets[       owner_name ][ network_name ] = {};
-   end
-
-   -- lua doesn't allow efficient counting here
-   local anz = 0;
-   for k in pairs( travelnet.targets[ owner_name ][ network_name ] ) do
-
-      if( k == station_name ) then
-         travelnet.show_message( pos, player_name, S("Error"),
-	    S("A station named '@1' already exists on this network. Please choose a different name!", station_name));
-         return;
-      end
-
-      anz = anz + 1;
-   end
-
-   -- we don't want too many stations in the same network because that would get confusing when displaying the targets
-   if( anz+1 > travelnet.MAX_STATIONS_PER_NETWORK ) then
-      travelnet.show_message( pos, player_name, S("Error"),
-	S("Network '@1', already contains the maximum number (@2) of allowed stations per network. "..
-	"Please choose a different/new network name.", network_name, travelnet.MAX_STATIONS_PER_NETWORK));
-      return;
-   end
-
-   -- add this station
-   travelnet.targets[ owner_name ][ network_name ][ station_name ] = {pos=pos, timestamp=os.time() };
-
-   -- do we have a new node to set up? (and are not just reading from a safefile?)
-   if( meta ) then
-
-      minetest.chat_send_player(player_name, S("Station '@1'" .." "..
-		"has been added to the network '@2'" ..
-		", which now consists of @3 station(s).", station_name, network_name, anz+1));
-
-      meta:set_string( "station_name",    station_name );
-      meta:set_string( "station_network", network_name );
-      meta:set_string( "owner",           owner_name );
-      meta:set_int( "timestamp",       travelnet.targets[ owner_name ][ network_name ][ station_name ].timestamp);
-
-      meta:set_string("formspec",
-                     "size[12,10]"..
-                     "field[0.3,0.6;6,0.7;station_name;"..S("Station:")..";"..
-										 minetest.formspec_escape(meta:get_string("station_name")).."]"..
-                     "field[0.3,3.6;6,0.7;station_network;"..S("Network:")..";"..
-										 minetest.formspec_escape(meta:get_string("station_network")).."]" );
-
-      -- display a list of all stations that can be reached from here
-      travelnet.update_formspec( pos, player_name, nil );
-
-      -- save the updated network data in a savefile over server restart
-      travelnet.save_data();
-   end
+travelnet.add_target = function(station_name, network_name, pos, player_name, meta, owner_name)
```

Comment:
Guess this is better. It's shorter and more understandable, IMO.
```suggestion
function travelnet.add_target(station_name, network_name, pos, player_name, meta, owner_name)
```

Same in other parts of the code.

Reply:
Are we still seeing line length issues? I thought it was at 120 chars, this line is < 100, and the checks are passing...

Though I agree with the change, I'm trying to resist the urge to fix anything that's not linting yet, to make sure the behaviour is 100% unchanged, and make it easier to review.

I will do PRs in the future to make the actual code saner

Reply:
Yep, no problems with line length.

Also, it's better to keep it under the 120 default than making luacheck ignore it. As the developer wiki puts it:
> If you need more you're probably doing something wrong.